src/java.base/share/classes/java/util/zip/Deflater.java

```diff
@@ -868,23 +886,44 @@ public void reset() {
     }
 
     /**
-     * Closes the compressor and discards any unprocessed input.
+     * Closes and releases the resources held by this {@code Deflater}
+     * and discards any unprocessed input.
+     * <p>
+     * If this method is invoked multiple times, the subsequent calls are treated as a no-op.
```

Comment:
I would say,

> If this method is invoked multiple times, the second and subsequent calls do nothing.

Reply:
Done. I've updated the end() methods javadoc of both `Deflater` and `Inflater` to use this recommended text.